deployments/automated-previews.mdx

```diff
@@ -151,59 +151,75 @@ jobs:
+in the Architect Cloud will be destroyed.
+
+<Warning>
+  This example assumes that the repo has `ARCHITECT_EMAIL`, `ARCHITECT_PASSWORD`, `ARCHITECT_ACCOUNT`, and `ARCHITECT_CLUSTER` set as CI/CD variables.
+</Warning>
 
 <Warning>
   `ARCHITECT_PASSWORD` must be a [personal access
   token](https://cloud.architect.io/login).
 </Warning>
 
 ```yaml
-# this example assumes that the repo has ARCHITECT_ACCOUNT and ARCHITECT_CLUSTER set as CI/CD variables
 stages:
   - preview
 
 variables:
-  ARCHITECT_ACCOUNT: <account-name>
-  ARCHITECT_CLUSTER: <cluster-name>
+  ARCHITECT_ACCOUNT: <set-this-as-a-CI/CD-variable>
+  ARCHITECT_CLUSTER: <set-this-as-a-CI/CD-variable>
   ARCHITECT_ENVIRONMENT: preview-$CI_MERGE_REQUEST_ID
+  ARCHITECT_CACHE_DIR: .architect-cache/
 
-default:
-  image: docker:latest
+.preview_common: &preview_common
+  image: docker:23.0.1
   services:
-    - docker:dind
+    - docker:23.0.1-dind
   before_script:
-    - apk add --update npm git
-    - apk add yq --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community
-    # Install architect cli and login
+    - apk add --update npm
     - npm install -g @architect-io/cli
-    - architect login -e $ARCHITECT_EMAIL -p $ARCHITECT_PASSWORD
+    - architect login
 
 deploy_preview:
+  <<: *preview_common
   stage: preview
-  script: |
-    architect environment:create $ARCHITECT_ENVIRONMENT --cluster $ARCHITECT_CLUSTER
-    architect deploy architect.yml --auto-approve --environment $ARCHITECT_ENVIRONMENT
+  timeout: 1h
+  script:
+    - architect environment:create "${ARCHITECT_ENVIRONMENT}" --description="https://gitlab.com/${CI_MERGE_REQUEST_PROJECT_PATH}/-/merge_requests/${CI_MERGE_REQUEST_IID}"
+    - architect deploy ./architect.yml --auto-approve --cache-directory=$ARCHITECT_CACHE_DIR
   environment:
-    name: architect/preview-$CI_MERGE_REQUEST_ID
-    url: https://cloud.architect.io/$ARCHITECT_ACCOUNT/environments/preview-$CI_MERGE_REQUEST_ID/
+    name: $ARCHITECT_ACCOUNT/preview-$CI_MERGE_REQUEST_IID
     on_stop: destroy_preview
   rules:
-    - if: $CI_MERGE_REQUEST_ID
+    - if: $CI_MERGE_REQUEST_IID
+  cache:
+    key: architect-preview-cache-$CI_COMMIT_REF_SLUG
+    paths:
+      - $ARCHITECT_CACHE_DIR
 
 destroy_preview:
+  <<: *preview_common
   stage: preview
-  variables:
-    ARCHITECT_ENVIRONMENT: preview-$CI_MERGE_REQUEST_ID
-  script: |
-    architect destroy --auto-approve --environment $ARCHITECT_ENVIRONMENT
-    architect environment:destroy --auto-approve $ARCHITECT_ENVIRONMENT
+  script:
+    - architect destroy --auto-approve
+    - architect environment:destroy "${ARCHITECT_ENVIRONMENT}" --auto-approve
   environment:
-    name: architect/preview-$CI_MERGE_REQUEST_ID
+    name: $ARCHITECT_ACCOUNT/preview-$CI_MERGE_REQUEST_IID
     action: stop
   rules:
     - if: $CI_MERGE_REQUEST_ID
+      allow_failure: true
       when: manual
```

Comment:
Can't this be automated on PR close?

Reply:
It is. This step just also gives the user a manual option, but we could remove it if you want

Reply:
Nope, all good - just wanted to ensure this is still automated and the user doesn't have to do anything